xbmc/addons/AddonManager.cpp

```diff
@@ -674,7 +677,18 @@ void CAddonMgr::FindAddons()
       SetChanged();
     }
   }
+
+  return result;
```

Comment:
You're right. I forgot to add a `result = true?  in the `if` block. Will remedy directly in master.